src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextJsonReader.cs

```diff
@@ -357,7 +358,10 @@ private List<Target> ReadTargets(ref Utf8JsonReader reader)
 
             while (reader.Read() && reader.IsTokenTypeProperty())
             {
-                targets.Add(ReadTarget(ref reader, reader.GetString()));
+                string? targetName = reader.GetString();
+                Debug.Assert(targetName != null);
```

Comment:
I don't think these `Debug.Assert`s are valid either. This data is coming from user input, there is no guarantee that these strings are not `null`.

Maybe the best approach here is to:

1. For this PR, just add the `!` to suppress the compiler warning.
2. Log a follow up issue to actually handle `null` being returned here. In a separate PR we can decide what to do when these strings are `null` (probably throw a decent exception), along with adding tests.